influent-transport/src/main/java/influent/internal/nio/NioChannelConfig.java

```diff
@@ -0,0 +1,129 @@
+package influent.internal.nio;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+public class NioChannelConfig {
+  public enum Protocol { TCP, TLS }
+  public enum TlsVersion {
+    None("None"),
+    TLSv1_1("TLSv1.1"),
+    TLSv1_2("TLSv1.2");
+    private final String version;
+
+    TlsVersion(String s) {
+      version = s;
+    }
+
+    @Override
+    public String toString() {
+      return version;
+    }
+  }
+
+  private boolean sslEnabled = false;
+  private String host;
+  private int port;
+  private String[] ciphers;
+  private SSLContext context;
+
+  public NioChannelConfig() {
+    sslEnabled = false;
+    context = null;
+  }
+
+  public NioChannelConfig(String host, int port, String protocol, String tlsVersion, String[] ciphers,
+                          String keystorePath, String keystorePassword, String keyPassword,
+                          String truststroePath, String truststrorePassword) {
+    this.host = host;
+    this.port = port;
+    this.ciphers = ciphers;
+    try {
+      if (protocol.equals("TLS")) {
+        sslEnabled = true;
+        context = SSLContext.getInstance(tlsVersion);
+        context.init(
+            createKeyManagers(keystorePath, keystorePassword, keyPassword),
+            createTrustManagers(truststroePath, truststrorePassword),
```

Comment:
No trust store is required when using a public root CA(or a CA trusted by root CA).
So if `truststorePath` is `null`, the second argument should be `null`.